qemu/tests/boot_nbdimage_with_qsd.py

```diff
@@ -0,0 +1,69 @@
+import os
+
+from virttest import data_dir
+from virttest import qemu_storage
+
+from avocado.utils import process
+from provider import qemu_img_utils as img_utils
+from provider.qsd import QsdDaemonDev
+
+
+def run(test, params, env):
+    """
+    Boot nbd image with qsd plus unix.
+
+    1) Prepare an installed OS image via converting
+    2) Export the OS image via QSD plus unix
+    3) Start VM from the exported image
+    """
+    def pre_test():
+        # prepare an installed OS image
+        test.log.info(
+            "Prepare an installed OS image file via converting, "
+            "and exporting it via QSD")
+        source = params["convert_source"]
+        target = params["convert_target"]
+        source_params = params.object_params(source)
+        target_params = params.object_params(target)
+        root_dir = data_dir.get_data_dir()
+        source_image = qemu_storage.QemuImg(source_params, root_dir, source)
+        target_image = qemu_storage.QemuImg(target_params, root_dir, target)
+
+        test.log.info("Convert %s to %s", source, target)
+        try:
+            source_image.convert(params, root_dir)
+        except process.CmdError as detail:
+            target_image.remove()
+            test.fail("Convert %s to %s failed for '%s', check please.",
```

Comment:
The values of three parameters are not output. FAIL: Convert %s to %s failed for '%s', check please. 
I think we should use "%" instead of "," here. Thanks.

Reply:
we may also:
try:
convert
finally:
target.remove

No need to add another error msg, for an exception will show the root cause of the converting failure

Reply:
We can not remove it after converting successfully for the converted image will be used as the exported image via QSD.

Reply:
oh yes, I lost my mind :(